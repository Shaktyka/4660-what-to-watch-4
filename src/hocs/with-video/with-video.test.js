import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';

import withVideo from './with-video.js';

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.node.isRequired,
};

const MockComponentWrapped = withVideo(MockComponent);

describe(`withVideo rendering`, () => {

  it(`withVideo renders correctly`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        src={` `}
        poster={` `}
        isPlaying={false}
        muted
      />
    ), {
      createNodeMock() {
        return {};
      }
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
