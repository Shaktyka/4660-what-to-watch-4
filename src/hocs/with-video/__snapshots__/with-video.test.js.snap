// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withVideo rendering withVideo renders correctly 1`] = `
<div>
  <video
    height="175"
    poster=" "
    type="video/webm"
    width="280"
  >
    Sorry, your browser doesn't support embedded video, but don't worry, you can 
    <a
      href=" "
    >
      download it
    </a>
    and watch it with your favorite video player!
  </video>
</div>
`;
